extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php

```diff
@@ -0,0 +1,44 @@
+<?php
+require_once( __DIR__ . '/../../../../maintenance/Maintenance.php' );
+
+class CleanupCommentsIndex extends Maintenance {
+
+	public function __construct() {
+		parent::__construct();
+
+		$this->addOption( 'dry-run', 'only checks how many rows is affected' );
+	}
+
+	public function execute() {
+		global $wgDBname;
+
+		$db = wfGetDB( DB_MASTER );
+
+		$ids = $db->selectFieldValues(
+			[ 'page', 'comments_index' ],
+			'comment_id',
+			[ 'page_id=comment_id', 'page_namespace not in (1,500,501,1200,1201,2000,2001)' ],
```

Comment:
In case of a glee it seems that NS_TALK entries are quite old:

```sql
mysql@geo-db-c-slave.query.consul[glee]>select count(*), page_namespace, max(rev_timestamp) from page, comments_index, revision where page_id=comment_id and page_latest = rev_id and page_namespace = 1;
+----------+----------------+--------------------+
| count(*) | page_namespace | max(rev_timestamp) |
+----------+----------------+--------------------+
|    50030 |              1 | 20150401112651     |
+----------+----------------+--------------------+
```

The same applies to NS 3 and 5.

Blog comments (501), Wall (1201) and Forum (2001) entries are fresh.

Let's remove NS_TALK entries as well.